test/lint/test_runner/src/main.rs

```diff
@@ -95,6 +95,85 @@ fs:: namespace, which has unsafe filesystem functions marked as deleted.
     }
 }
 
+/// Return the pathspecs for whitespace related excludes
+fn get_pathspecs_exclude_whitespace() -> Vec<String> {
+    let mut list = get_pathspecs_exclude_subtrees();
+    list.extend(
+        [
+            // Permanent excludes
+            "*.patch",
+            "src/qt/locale",
+            "contrib/windeploy/win-codesign.cert",
+            "doc/README_windows.txt",
+            // Temporary excludes, or existing violations
+            "doc/release-notes/release-notes-0.*",
+            "contrib/init/bitcoind.openrc",
+            "contrib/macdeploy/macdeployqtplus",
+            "src/crypto/sha256_sse4.cpp",
+            "src/qt/res/src/*.svg",
+            "test/functional/test_framework/crypto/ellswift_decode_test_vectors.csv",
+            "test/functional/test_framework/crypto/xswiftec_inv_test_vectors.csv",
+            "contrib/qos/tc.sh",
+            "contrib/verify-commits/gpg.sh",
+            "src/univalue/include/univalue_escapes.h",
+            "src/univalue/test/object.cpp",
+            "test/lint/git-subtree-check.sh",
+        ]
+        .iter()
+        .map(|s| format!(":(exclude){}", s)),
+    );
+    list
+}
+
+fn lint_trailing_whitespace() -> LintResult {
+    let trailing_space = git()
+        .args(["grep", "-I", "--line-number", "\\s$", "--"])
+        .args(get_pathspecs_exclude_whitespace())
+        .status()
+        .expect("command error")
+        .success();
+    if trailing_space {
+        Err(r#"
```

Comment:
How about displaying the offending lines with something like:
```diff
diff --git a/test/lint/test_runner/src/main.rs b/test/lint/test_runner/src/main.rs
index cffb477950..6f16620605 100644
--- a/test/lint/test_runner/src/main.rs
+++ b/test/lint/test_runner/src/main.rs
@@ -8 +8 @@ use std::path::{Path, PathBuf};
-use std::process::{Command, Stdio};
+use std::process::Command;
@@ -129 +129 @@ fn lint_trailing_whitespace() -> LintResult {
-    let output = git()
+    let output = check_output(git()
@@ -131,6 +131,4 @@ fn lint_trailing_whitespace() -> LintResult {
-        .args(get_pathspecs_exclude_whitespace())
-        .output()
-        .expect("command error");
-    let stderr = String::from_utf8_lossy(&output.stderr);
-    if !stderr.is_empty() {
-        Err(r#"
+        .args(get_pathspecs_exclude_whitespace()));
+    if let Ok(message) = output {
+        Err(format!(r#"
+{}
@@ -146,2 +144,2 @@ sourced files to the exclude list.
-            "#
-        .to_string())
+            "#,
+        message))
@@ -150 +148 @@ sourced files to the exclude list.
-    }    
+    }
@@ -154 +152 @@ fn lint_tabs_whitespace() -> LintResult {
-    let tabs = git()
+    let output= check_output(git()
@@ -157,6 +155,5 @@ fn lint_tabs_whitespace() -> LintResult {
-        .args(get_pathspecs_exclude_whitespace())
-        .status()
-        .expect("command error")
-        .success();
-    if tabs {
-        Err(r#"
+        .args(get_pathspecs_exclude_whitespace()));
+
+    if let Ok(message) = output {
+        Err(format!(r#"
+{}
@@ -170,2 +167,2 @@ Please add any false positives, such as subtrees, or externally sourced files to
-            "#
-        .to_string())
+            "#,
+        message))
```

Reply:
Ah, this is git paging the output on my system if the output is not piped to somewhere else. I'll just switch the paging off for `git grep` then, it's annoying anyway.

Reply:
Or just `git --no-pager`?